models/orderDao.js

```diff
@@ -0,0 +1,103 @@
+const appDataSource = require('./appDataSource');
+const { OrderStatuses, SeatClass } = require('./enum');
+const { CustomError } = require('../utils/error');
+
+const createOrder = async (totalAmount, userId, musicalId, seatArray, musicalScheduleId) => {
+  const queryRunner = appDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+  const personnel = seatArray.length;
+  try {
+    const getMusicalDateId = await queryRunner.query(
+      `SELECT musical_date.id
+      FROM musical_date
+      JOIN musical_schedules ON musical_date.id = musical_schedules.musical_date_id
+      JOIN theaters ON musical_schedules.theater_id = theaters.id
+      JOIN musicals ON musical_schedules.musical_id = musicals.id
+      WHERE musicals.id = ? AND musical_schedules.id = ?
+      `,
+      [musicalId, musicalScheduleId]
+    );
+    const musicalDateId = getMusicalDateId[0].id;
+    await queryRunner.query(
+      `INSERT INTO orders(
+        personnel,
+        total_amount,
+        user_id,
+        musical_id,
+        musical_schedule_id,
+        order_status_id
+        )
+        SELECT ?, ?, ?, musicals.id, musical_schedules.id, ${OrderStatuses.결제완료}
+        FROM musicals
+        JOIN musical_schedules ON musical_schedules.musical_id = musicals.id
+        JOIN musical_date ON musical_date.id = musical_schedules.musical_date_id
+        WHERE musicals.id = ? AND musical_date.id = ?
+      `,
+      [personnel, totalAmount, userId, musicalId, musicalDateId]
+    );
+
+    const gerOrderId = await queryRunner.query(
+      `SELECT
+      id
+      FROM orders
+      WHERE orders.user_id = ?
+      ORDER BY orders.id DESC
+      LIMIT 1
+      `,
+      [userId]
+    );
+    const orderId = gerOrderId[0].id;
```

Comment:
```suggestion
const [{ id: orderId }] = await queryRunner.query(
  `SELECT
  id
  FROM orders
  WHERE orders.user_id = ?
  ORDER BY orders.id DESC
  LIMIT 1
  `,
  [userId]
);
```
- 가독성을 위해 위와 같이 바꾸어 보는 것은 어떨까요?

Reply:
- 위 내용이 정상 작동한다면 다른 변수에서도 한 번 고민해서 반영을 고민해보세요!